### PR TITLE
Gemify

### PR DESCRIPTION
**Pull request by lukeasrodgers**
Converts this repo into a gem.

Still working on integration with our rails app, making sure migration generation works, etc.

### CODE REVIEW

**Comment by lukeasrodgers** [comment]
@madnoodlesmlin I don't think we are actually using the `json` gem anywhere. Can we remove it as a dependency? Or maybe just make it a development dependency, if necessary?

**Comment by madnoodlesmlin** [comment]
@lukeasrodgers since orientdb4r gem uses json 1.7.5 and on my machine I have multiple versions of json gem and by default it uses version 1.8.1, I just specify it so there will not have any conflict.

**Comment by lukeasrodgers** [comment]
@madnoodlesmlin I don't see anywhere that orientdb4r requires json 1.7.5. Actually, their use of it seems a bit odd: they have a commented out statement requiring 1.5.1, so I guess they're actually just using ruby's built-in json module.

**Comment by madnoodlesmlin** [comment]
@lukeasrodgers uh I see.

File: lib/orientdb_schema_migrator/migration.rb [hunk 1]
```diff
@@ -40,33 +40,31 @@ def self.rename_class old_name, new_name
...
 
     def self.drop_property class_name, property_name
+      assert_class_exists!(class_name)
+      return false unless property_exists?(class_name, property_name)
       ODBClient.command "drop property #{class_name}.#{property_name}"
     end
 
     def self.alter_property class_name, property_name, attribute_name, new_value
+      assert_class_exists!(class_name)
+      return false unless property_exists?(class_name, property_name)
```
Comment (madnoodlesmlin) [inline lib/orientdb_schema_migrator/migration.rb]: I think we might want to change all validation check to use methods
assert_class_exists!(class_name)
assert_property_exists!(class_name, property_name)
Is there any reason that we want this method to return false?

Comment (lukeasrodgers) [inline lib/orientdb_schema_migrator/migration.rb]: Ah no, you're right, I'll fix that.

File: lib/orientdb_schema_migrator/migration.rb [hunk 2]
```diff
@@ -40,33 +40,31 @@ def self.rename_class old_name, new_name
         ODBClient.command "alter class #{old_name} name #{new_name}"
         return true
       else
-        puts "#{class_name} doesn't exist"
         return false
```
Comment (madnoodlesmlin) [inline lib/orientdb_schema_migrator/migration.rb]: do we want to change line 5,22,and here to use assert_class_exists! method?

File: lib/orientdb_schema_migrator/migration.rb [hunk 3]
```diff
@@ -40,33 +40,31 @@ def self.rename_class old_name, new_name
...
-      if property_exists? class_name, property_name
-        ODBClient.command "create index #{index_name} on #{class_name} (#{property_name}) #{type}"
-        return true
-      else
-        puts "please create property #{property_name} first"
-        return false
-      end
+    def self.add_index class_name, property_name, index_name, type
+      assert_class_exists!(class_name)
+      assert_property_exists!(class_name, property_name)
```
Comment (akachrislee) [inline lib/orientdb_schema_migrator/migration.rb]: does ODB complain if we're trying to add an index that already exists?
i.e. should we be checking `index_exists?` before creating the index?

File: lib/orientdb_schema_migrator/migration.rb [hunk 4]
```diff
@@ -40,33 +40,31 @@ def self.rename_class old_name, new_name
...
+      ODBClient.command "create index #{index_name} on #{class_name} (#{property_name}) #{type}"
     end
 
     def self.drop_index index_name
```
Comment (akachrislee) [inline lib/orientdb_schema_migrator/migration.rb]: does ODB complain if we're trying to remove an index that DOESN'T exists?
i.e. should we be checking `index_exists?` before removing the index?

Comment (lukeasrodgers) [inline lib/orientdb_schema_migrator/migration.rb]: Good question(s) -- I think ODB doesn't care in these situations, but I will double check.

Comment (madnoodlesmlin) [inline lib/orientdb_schema_migrator/migration.rb]: I think we should add a check for index when we create index. We don't have
to check when we drop index.

File: lib/orientdb_schema_migrator/migration.rb [hunk 5]
```diff
@@ -40,33 +40,31 @@ def self.rename_class old_name, new_name
...
         return true
       else
-        puts "#{class_name} doesn't exist"
         return false
       end
     end
 
-    def self.add_property class_name, property_name, type,property_options={}
+    def self.add_property class_name, property_name, type, property_options={}
+      assert_class_exists!(class_name)
```
Comment (madnoodlesmlin) [inline lib/orientdb_schema_migrator/migration.rb]: @lukeasrodgers Do we want to check if property already exists?

**Comment by lukeasrodgers** [comment]
@madnoodlesmlin those changes look good to me.

**Comment by lukeasrodgers** [comment]
@madnoodlesmlin do you wanna review my latest changes? I think if they look okay, this is ready to merge. Probably not perfect/missing features, but good enough, I think.

**Comment by madnoodlesmlin** [comment]
looks good. :+1:

**Pull request merged**